value.go

```diff
@@ -56,3 +58,54 @@ func getValue(info TypeInfo, v mapping.Value) (any, error) {
 		return nil, unsupportedTypeError(typeToStringMap[t])
 	}
 }
+
+func createValue(lt mapping.LogicalType, v any) (*mapping.Value, error) {
+	var vv mapping.Value
+	var err error
+	t := Type(mapping.GetTypeId(lt))
+	switch t {
+	case TYPE_BOOLEAN:
+		vv, err = mapping.CreateBool(v.(bool)), nil
+	case TYPE_TINYINT:
+		vv, err = mapping.CreateInt8(v.(int8)), nil
+	case TYPE_SMALLINT:
+		vv, err = mapping.CreateInt16(v.(int16)), nil
+	case TYPE_INTEGER:
+		// TODO: do all int types need this casting?
+		if i, ok := v.(int); ok {
+			vv, err = mapping.CreateInt32(int32(i)), nil
+		} else {
+			vv, err = mapping.CreateInt32(v.(int32)), nil
+		}
```

Comment:
We should be able to reduce this to `vv, err = mapping.CreateInt32(v.(int32)), nil`.

Reply:
That's the expected error, if the Go type does not match the expected bind type.

Adjusting the test like so: `map[string]any{"v": "baz", "i": int32(42)},` (adding the cast) solves this.
We get a similar failure when, for example, passing a boolean instead of a string: `map[string]any{"v": false, "i": int32(42)},`.

We might want to properly solve this in the future by not just assuming that the cast will succeed everywhere (`mapping.CreateBool(v.(bool))`, ...). Instead, we could use `i, ok := v.(type)` to throw better error messages for all cases...